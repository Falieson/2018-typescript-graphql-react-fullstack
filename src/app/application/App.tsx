// import './fonts'

import * as React from 'react'

import Page from '../components/page/'
import Routes from '../routes/'

export default function App() {
  return <Page>
    <Routes />
  </Page>
}
