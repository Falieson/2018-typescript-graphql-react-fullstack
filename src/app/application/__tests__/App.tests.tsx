import { shallow } from 'enzyme'
import * as React from 'react'
import * as ReactDOM from 'react-dom'
import App from '../'

describe('Application Integration', () => {
  it('loads w/o crashing', () => {
    const root = document.createElement('div')
    root.setAttribute('id', 'root')
    ReactDOM.render(<App />, root)
  })
  it('matches the snapshot', () => {
    const wrapper = shallow(<App />)
    expect(wrapper).toMatchSnapshot()
  })
})
