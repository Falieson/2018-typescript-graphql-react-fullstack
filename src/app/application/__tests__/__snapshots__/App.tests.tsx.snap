// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Application Integration matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <default_1 />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <App />,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <App />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
